zulipterminal/ui_tools/boxes.py

```diff
@@ -457,12 +457,23 @@ def autocomplete_users(
         )
 
         user_names = [user["full_name"] for user in sorted_matching_users]
+
+        user_names_counter = Counter(user_names)
```

Comment:
It's worth adding a note/todo/fixme here as to why we use this - this definitely seems faster and more efficient than the previous implementation, but it's worth noting why, and how/why we could improve this in future (we generate this counter each time we do an autocomplete?)